src/main/java/com/google/devtools/build/lib/bazel/rules/python/BazelPyRuleClasses.java

```diff
@@ -221,6 +223,12 @@ responsible for creating (possibly empty) __init__.py files and adding them to t
           .add(
               attr("$py_toolchain_type", NODEP_LABEL)
                   .value(env.getToolsLabel("//tools/python:toolchain_type")))
+          /* Only used when no py_runtime() is available.
+          (when --incompatible_use_python_toolchains=false --python_path is used on Windows)
+          */
```

Comment:
Lets mention https://github.com/bazelbuild/bazel/issues/7901 here. That seems to be the best root issue that describes the edge case of when no py_runtime is available.